frontend/src/pages/protocol-token/list/all.vue

```diff
@@ -0,0 +1,274 @@
+											:value="String(token.totalMinted || '0')"
+											:decimals="token.decimal || 0"
+											:fixed-decimals="2"
+											:format-number="true"
+										/>
+									</div>
+								</td>
+								<td class="py-4 px-4">
+									<div class="font-medium">
+										<PltAmount
+											:value="String(token.totalBurned || '0')"
+											:decimals="token.decimal || 0"
+											:fixed-decimals="2"
+											:format-number="true"
+										/>
+									</div>
+								</td>
+								<td class="py-4 px-4">
+									<span class="font-medium">
+										{{ token.totalUniqueHolders || 0 }}
+									</span>
+								</td>
+								<td class="py-4 px-4">
+									<AccountLink
+										v-if="token.issuer?.asString"
+										:address="token.issuer.asString"
+									/>
+									<span v-else class="text-theme-text-secondary text-sm"
+										>N/A</span
+									>
+								</td>
+								<td class="py-4 px-4">
+									<Tooltip
+										v-if="token.block?.blockSlotTime"
+										:text="formatTimestamp(token.block.blockSlotTime)"
+									>
+										<span class="text-sm text-theme-text-secondary">
+											{{
+												convertTimestampToRelative(
+													token.block.blockSlotTime,
+													NOW
+												)
+											}}
+										</span>
+									</Tooltip>
+									<span v-else class="text-sm text-theme-text-secondary"
+										>N/A</span
+									>
+								</td>
+							</TableRow>
+						</tbody>
+					</table>
+				</div>
+			</div>
+
+			<Pagination
+				v-if="pageInfo"
+				:page-info="pageInfo"
+				:go-to-page="goToPage"
+			/>
+		</div>
+	</div>
+</template>
+
+<script lang="ts" setup>
+import { ref, watch } from 'vue'
+import { usePltTokensPagedQuery } from '~/queries/usePltTokensPagedQuery'
+import { usePagination } from '~/composables/usePagination'
+import { useDateNow } from '~/composables/useDateNow'
+import { formatTimestamp, convertTimestampToRelative } from '~/utils/format'
+
+definePageMeta({
+	middleware: 'plt-features-guard',
+})
+
+const { NOW } = useDateNow()
+
+const pageSize = 25
+
+const { after, before, first, last, goToPage } = usePagination({
+	pageSize,
+})
+
+const currentPage = ref(1)
+
+watch([after, before], () => {
+	if (!after.value && !before.value) {
+		currentPage.value = 1
+	} else if (after.value) {
+		currentPage.value = currentPage.value + 1
+	} else if (before.value) {
+		currentPage.value = Math.max(1, currentPage.value - 1)
+	}
+})
+
+const {
+	data,
+	pageInfo,
+	loading: pltTokenLoading,
+} = usePltTokensPagedQuery({
+	first,
+	last,
+	after,
+	before,
+})
+
+const pltTokenData = ref(data.value)
+
+watch(
+	() => data.value,
+	newData => {
+		pltTokenData.value = newData
+	},
+	{ immediate: true }
+)
+
+const truncateAddress = (address: string, length: number = 12) => {
+	if (!address) return ''
+	if (address.length <= length) return address
+	return `${address.slice(0, length / 2)}...${address.slice(-length / 2)}`
+}
```

Comment:
Please add doc comments to a function and move this function to a general place is it can/should likely be re-used (not strictly related to PLT tokens). 

Please explain the `length` parameter in the function.